sql-cli/tests/test_dag_generator.py

```diff
@@ -97,6 +173,58 @@ def test_generate_dag_invalid_directory(root_directory, dags_directory, generate
 """
 
 
+@freeze_time(datetime.datetime(2016, 12, 22, 3, 49, 0))
 def test_generate_dag_black_compliant(root_directory, dags_directory):
     dag_file = generate_dag(directory=root_directory, dags_directory=dags_directory, generate_tasks=True)
-    assert dag_file.read_text() == compliant_dag
+    computed = dag_file.read_text()
+
+    first_part = compliant_dag_first_part
+    if airflow_version() < Version("2.4"):
+        first_part = compliant_dag_first_part.replace("schedule", "schedule_interval")
```

Comment:
I am not a fan of calling `.replace` 🙈 . It would be nicer if we had `compliant_dag` and `compliant_dag_latest` and create two tests and based on the airflow version skip one of them.

Reply:
Or what would be probably even better would be to parametrize the test and mock the airflow version dynamically.

Reply:
I will take a stab at it.

Reply:
I approved the PR, and will improve the test separately to unblock 0.3. release.

Reply:
Agreed with @feluelle we'll improve the test in a separate PR.